packages/url/src/clean-for-slug.js

```diff
@@ -23,11 +23,9 @@ export function cleanForSlug( string ) {
 	if ( ! string ) {
 		return '';
 	}
-	return trim(
-		deburr( string )
-			.replace( /[\s\./]+/g, '-' )
-			.replace( /[^\p{L}\p{N}_-]+/gu, '' )
-			.toLowerCase(),
-		'-'
-	);
+	return removeAccents( string )
+		.replace( /[\s\./]+/g, '-' )
```

Comment:
I'd love it for a follow up maybe to either abstract the regex here into a named variable or alternatively adding a comment above to explain what each of these are doing :) Right now it is hard to read whats happening

Reply:
I agree 👍 I've seen many folks struggle with remembering the unicode categories. 

Also, adding comments at this stage of the PR is pretty cheap 😉  Added in https://github.com/WordPress/gutenberg/pull/41687/commits/8832905853a7bafd75d6c8d71400e65120d92322